utils.go

```diff
@@ -386,6 +386,9 @@ func ReadStringWithMax(r io.Reader, maxLength uint64) (string, error) {
 	}
 
 	bufp := stringBufPool.Get().(*[]byte)
+	if cap(*bufp) < int(l) {
+		*bufp = append((*bufp)[:cap(*bufp)], make([]byte, int(l)-cap(*bufp))...)
```

Comment:
This is going to replace the string in the buffer pool (with an unnecessary copy of bytes we don't care about). Shouldn't we instead:

1. Detect that we're requesting a string that's longer than the max size in the buffer pool.
2. Side-step the buffer-pool.

That or just use https://github.com/libp2p/go-buffer-pool/ (with a package-local pool, probably).

Reply:
I guess the problem is that if we have a larger-than-default max string size, then we're going to have problems with the buffer pool here. Not entirely sure the best way to deal with that on a per-user basis since the buffer pool is a global

Reply:
side-stepping the buffer pool doesnt seem right, as the choice of max string length is somewhat arbitrary and the buffer pool should work with it

Reply:
In that case, I'd just use the go-buffer-pool package from libp2p.

Reply:
On the other hand, the use of these buffers is pretty short-lived, so I don't expect we'll have a lot of them in the pool. I guess replacing smaller ones with larger ones kind of makes sense.

Reply:
yeah i thought it seemed fine to extend the sizes, its basically just increasing the pool size to whatever max string length is